src/Titanium.Web.Proxy/Helpers/HttpWriter.cs

```diff
@@ -67,11 +67,15 @@ private Task writeAsyncInternal(string value, bool addNewLine, CancellationToken
                         idx += newLineChars;
                     }
 
-                    return stream.WriteAsync(buffer, 0, idx, cancellationToken);
+                    return stream.WriteAsync(buffer, 0, idx, cancellationToken).ContinueWith((antecedent) =>
```

Comment:
Good catch. I think an await here would suffice though, instead of a continueWith.